tests/unit/processor/generic_resolver/test_generic_resolver.py

```diff
@@ -409,3 +467,163 @@ def test_resolve_generic_and_multiple_match_first_only(self):
         self.object.process(document)
 
         assert document == expected
+
+    def test_resolve_from_cache_with_large_enough_cache(self):
+        config = deepcopy(self.CONFIG)
+        config["max_cache_entries"] = 10
+        self.object = Factory.create({"generic_resolver": config})
+
+        rule_dict = {
+            "filter": "to_resolve",
+            "generic_resolver": {
+                "field_mapping": {"to_resolve": "resolved"},
+                "resolve_list": {".+ar": "res_bar", ".+oo": "res_foo"},
+            },
+        }
+        event = {"to_resolve": "foo"}
+        self._load_specific_rule(rule_dict)
+        self.object.setup()
+
+        self.object.metrics.new_results = 0
+        self.object.metrics.cached_results = 0
+        self.object.metrics.num_cache_entries = 0
+
+        self.object.process(event)
+
+        assert self.object.metrics.new_results == 1
+        assert self.object.metrics.cached_results == 0
+        assert self.object.metrics.num_cache_entries == 1
+
+        self.object.process(event)
+
+        assert self.object.metrics.new_results == 2
+        assert self.object.metrics.cached_results == 1
+        assert self.object.metrics.num_cache_entries == 2
```

Comment:
with my current understanding I would have expected that `num_cache_entries` is only 1 here, as you pass in the same event two times? also `new_results` I thought would be `1`, can you maybe explain the metrics to me again?

Reply:
Those integers don't represent the actual values of the metrics, however they are calculated based on them.

The values have to be added to the metrics instead of setting them for the metrics to work:

```python
self.metrics.new_results += cache_info.misses
self.metrics.cached_results += cache_info.hits
self.metrics.num_cache_entries += cache_info.currsize
```

The integers seem to represent the sum of values that have been added.
For passing the same event twice it is:

```
new_results += 1_miss  # 0 += 1
new_results += 1_miss  # 1 += 1
=> 2
```

For passing two different events it is:

```
new_results += 1_miss    # 0 += 1
new_results += 2_misses  # 1 += 2
=> 3
```

Reply:
ahh wait, sorry now I understand. In the tests we mock the `GaugeMetric` again with a regular `int` that's why the value continues increasing. I missed that. Could you then maybe add a comment to the test describing that the difference between the metric values (before and after each `self.object.process` call) is actually the relevant information because of the "mock"?

Reply:
I have added docstrings to the relevant tests.